core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -765,7 +765,29 @@ class WorkspaceService(protected val userInfo: UserInfo,
               )
             }
           }
-          workspaces.map { workspace =>
+
+        // Internal folks may create MCWorkspaces in local WorkspaceManager instances, and those will not
+        // be reachable when running against the dev environment.
+        val reachableWorkspaces = workspaces.filter { workspace =>
+          workspace.workspaceType match {
+            case WorkspaceType.McWorkspace =>
+              try {
+                workspaceManagerDAO.getWorkspace(workspace.workspaceIdAsUUID, userInfo.accessToken)
```

Comment:
Any opinions about avoiding 2 calls to `getWorkspace` per McWorkspace? It would require either storing the first result from `getWorkspace` or doing a combination of filter/map. I couldn't figure out how to do the exception handling with a collect, but I thought Scala experts might have some ideas.

On the other hand, I would expect that the getWorkspace call is O(n), and thus calling it twice shouldn't introduce much inefficiency. @aherbst-broad am I incorrect about the cost of calling getWorkspace?

Reply:
list.mapFilter() uses a function that returns an option, and filters out all None values.

Reply:
Ohhh, that sounds perfect. @yunip20, can you give that a try?